core/src/main/scala/scalaz/std/String.scala

```diff
@@ -38,33 +38,32 @@ trait StringFunctions {
   def charsNel(s:String, e: => NonEmptyList[Char]) : NonEmptyList[Char] = charsNel(s) getOrElse e
 
   // Parsing functions.
+  private def intChars(s: String): Boolean = s.forall(c => c.isDigit || c == '-')
+  private def floatChars(s: String): Boolean = s.forall(c => c.isDigit || c == '-' || c == '.')
 
-  def parseBoolean(s:String): Validation[IllegalArgumentException, Boolean] =
-    Validation.fromTryCatchThrowable[Boolean, IllegalArgumentException](s.toBoolean)
+  private def asNumber[T](f: String => T, b: String => Boolean, r: Tuple2[T, T], s: String, t: String): Validation[String, T] =
+    try {
+      Success(f(s))
+    } catch {
+      case _: NumberFormatException if intChars(s) => Failure(s"${s} is outside of range for ${t} (${r._1} - ${r._2})")
+      case _: NumberFormatException => Failure(s"${s} does not represent a valid ${t}")
+      case e: Throwable => Failure(e.getMessage)
```

Comment:
apparently `NonFatal` is better... (personally I think if you're making decisions based on whether you catch or don't catch OOM, you're fucked anyway)